### DISCUSSION TITLE
[Help Wanted] How to properly clean up previous selection?

### DISCUSSION THREAD

**Discussion by wzhchin** [xournalpp/xournalpp#3325] [General] [answered]
I've recently been trying to add pdf text selection support to Xournalpp, and this issue is related to [that](https://github.com/xournalpp/xournalpp/issues/1745).

This week, I'm trying to add the ability to select paragraphs and switch between and within regions. Selecting only content within a region is useful in some places, such as selecting text within a table or some two-column pdfs. (But the selecting text within a region part is not finished yet, so please disregard that, I'm going to continue after I solve this problem below).

But as still do not know much about cairo, I encountered a problem, can not clear the previous selection block, I have made some attempts, but still can not get it to work properly.

https://user-images.githubusercontent.com/51843252/129227387-d6fdb7de-d162-476d-bf46-83de236b9fbf.mp4

How can I modify my code to make it work properly? Any suggestions or hints?

related code: https://github.com/aeghn/xournalpp/blob/445dbb3c0bbe11e67037815ce51e09ce3270c2f7/src/control/PDFTextSelectControl.cpp#L161

Thanks in advance.

**Comment by Technius** [comment, marked as answer]
It looks like you're missing a call to `view->rerenderRect` or `view->rerenderElement` (where `view` is the `XojPageView*`) after you select the text, so the highlight is being written to the old view buffer.

I would advise you to submit a draft pull request (no matter how unfinished/ugly your code is!), which will help us maintainers give you more useful feedback and answer any questions that may come up as you work on this.
